### PR TITLE
Only pass revision argument to diff command when required

### PR DESCRIPTION
**Pull request by vhda**
Move common diff generation code to local script function.

### CODE REVIEW

**Comment by vhda** [comment]
@blueyed There you go. This code is a bit cleaner than what I did this morning.

File: autoload/quickfixsigns/vcsdiff.vim [hunk 1]
```diff
@@ -573,6 +555,33 @@ function! s:GetParam(name, type, default) abort "{{{3
 endf
 
 
+function! s:Diff(filename, vcs_type) "{{{3
+  let cmdt    = s:Config(a:vcs_type).cmd
+  let rev_arg = s:GetParam('quickfixsigns#vcsdiff#rev_arg', a:vcs_type, s:Config(a:vcs_type).rev_arg)
```
Comment (tomtom) [inline autoload/quickfixsigns/vcsdiff.vim]: rev_arg isn't a (user-configurable) "parameter" but a fixed value.

File: autoload/quickfixsigns/vcsdiff.vim [hunk 2]
```diff
@@ -573,6 +555,33 @@ function! s:GetParam(name, type, default) abort "{{{3
...
+  let extra   = s:GetParam('quickfixsigns#vcsdiff#extra_args', a:vcs_type, '')
+  let dir     = fnamemodify(a:filename, ':h')
+  let file    = fnamemodify(a:filename, ':t')
+
+  " Create command string
+  let cmds = printf("%s %s", g:quickfixsigns#vcsdiff#cd, shellescape(dir))
+  if exists('g:quickfixsigns#vcsdiff#revision') && !empty('g:quickfixsigns#vcsdiff#revision')
```
Comment (tomtom) [inline autoload/quickfixsigns/vcsdiff.vim]: g:quickfixsigns#vcsdiff#revision should be queried as user-configurable parameter.

I'd rather set revision to the formatted revision argument if a revision is specified or an empty string if no rev is specified. This makes the code duplication in the else branch obsolete.

File: doc/quickfixsigns.txt [hunk 3]
```diff
@@ -437,7 +437,7 @@ g:quickfixsigns#vcsdiff#vcs    (default: {...})
         cmd ... command templates that generate a unified diff file. 
         "%s" is replaced with the filename.
         dir ... the directory name
-        revision ... The default revision/branch
+        rev_arg ... argument to selection revision to diff against
```
Comment (tomtom) [inline doc/quickfixsigns.txt]: Thanks for updating the help file. The help file is auto-generated from time to time so it's not absolutely necessary.

File: autoload/quickfixsigns/vcsdiff.vim (hunk 1 above)
Comment (vhda) [inline autoload/quickfixsigns/vcsdiff.vim]: Do you mean that I should do this?

```Vim
let rev_arg = s:Config(a:vcs_type).rev_arg
```

File: autoload/quickfixsigns/vcsdiff.vim (hunk 2 above)
Comment (vhda) [inline autoload/quickfixsigns/vcsdiff.vim]: Great idea!

File: doc/quickfixsigns.txt (hunk 3 above)
Comment (vhda) [inline doc/quickfixsigns.txt]: What do you use for that purpose? I could benefit from such a flow.

File: autoload/quickfixsigns/vcsdiff.vim (hunk 1 above)
Comment (tomtom) [inline autoload/quickfixsigns/vcsdiff.vim]: Yes.

File: doc/quickfixsigns.txt (hunk 3 above)
Comment (tomtom) [inline doc/quickfixsigns.txt]: I use this ruby script: https://github.com/tomtom/vimtlib/tree/master/ruby

**Comment by vhda** [comment]
Updated.

**Comment by tomtom** [comment]
Thanks!

**Comment by vhda** [comment]
Thank **you** for your helpful suggestions!

**Comment by blueyed** [comment]
Thanks to both of you from me! :)

**Pull request merged**